### PR TITLE
Planck Layout Macro Refactor, Part II

### PR DESCRIPTION
**Pull request by noroadsleft**
My previous attempt at fixing the Planck's codebase did not succeed in repairing all the issues, so I took another run at it, and also drafted some help.

### CODE REVIEW

File: keyboards/planck/ez/info.json [hunk 1]
```diff
@@ -0,0 +1,115 @@
+{
+  "keyboard_name": "Planck EZ",
+  "keyboard_folder": "planck/ez",
+  "url": "https://olkb.com/planck",
```
Comment (jackhumbert) [inline keyboards/planck/ez/info.json]: We can have this go to https://ergodox-ez.com/pages/planck for now - I think we'll have a better URL for it later :)

File: keyboards/planck/ez/info.json [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+        { "x": 4, "y": 3 },
+        { "x": 5, "y": 3, "w": 2 },
+        { "x": 7, "y": 3 },
+        { "x": 8, "y": 3 },
+        { "x": 9, "y": 3 },
+        { "x": 10, "y": 3 },
+        { "x": 11, "y": 3 }
+      ]
+    },
+    "LAYOUT_ortho_4x12": {
```
[the remark this answers is not in the document]
Comment (drashna) [reply keyboards/planck/ez/info.json]: Also, the music map needs that layout, or ... we need to move the map to each keyboard, individually. 

Basically, there are a number of reasons why we need it... and the entire planck director is starting to become a pain to manage.

File: keyboards/planck/ez/info.json (hunk 1 above)
Comment (drashna) [reply keyboards/planck/ez/info.json]:
```suggestion
  "url": "https://ergodox-ez.com/pages/planck",
```

File: keyboards/planck/ez/info.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (noroadsleft) [reply keyboards/planck/ez/info.json]: > Given that the default uses this, I think I answered my own question, nvm :)

This is pretty much exactly the reason. The default keymap and the music map both use `LAYOUT_planck_grid`. Configurator's UI code still visibly renders a key even if you set its dimensions to 0x0, and there's no way in the API to exclude a layout on a per-keyboard basis.

**Comment by drashna** [comment]
Thanks, and tested, just in case.   Looks good!

**Pull request merged**